feature/platform/transceiver/tests/zr_low_power_mode_test/zr_low_power_mode_test.go

```diff
@@ -0,0 +1,174 @@
+			defer streamMfgName.Close()
+			streamMfgDate := samplestream.New(t, dut, gnmi.OC().Component(tr).MfgDate().State(), samplingInterval)
+			defer streamMfgDate.Close()
+			streamHwVersion := samplestream.New(t, dut, gnmi.OC().Component(tr).HardwareVersion().State(), samplingInterval)
+			defer streamHwVersion.Close()
+			streamFirmwareVersion := samplestream.New(t, dut, gnmi.OC().Component(tr).FirmwareVersion().State(), samplingInterval)
+			defer streamFirmwareVersion.Close()
+
+			allStream := map[string]*samplestream.SampleStream[string]{
+				"serialNo":        streamSerialNo,
+				"partNo":          streamPartNo,
+				"description":     streamDescription,
+				"mfgName":         streamMfgName,
+				"mfgDate":         streamMfgDate,
+				"hwVersion":       streamHwVersion,
+				"firmwareVersion": streamFirmwareVersion,
+			}
+			validateStreamOutput(t, dut, allStream)
+
+			opInst := samplestream.New(t, dut, gnmi.OC().Component(tr).OpticalChannel().OutputPower().Instant().State(), samplingInterval)
+			defer opInst.Close()
+			if opInstN := opInst.Next(t); opInstN != nil {
+				if _, ok := opInstN.Val(); ok {
+					t.Fatalf("streaming /components/component/optical-channel/state/output-power/instant is not expected to be reported")
+				}
+			}
+
+			opAvg := samplestream.New(t, dut, gnmi.OC().Component(tr).OpticalChannel().OutputPower().Avg().State(), samplingInterval)
+			defer opAvg.Close()
+			if opAvgN := opAvg.Next(t); opAvgN != nil {
+				if _, ok := opAvgN.Val(); ok {
+					t.Fatalf("streaming /components/component/optical-channel/state/output-power/avg is not expected to be reported")
+				}
+			}
+
+			opMin := samplestream.New(t, dut, gnmi.OC().Component(tr).OpticalChannel().OutputPower().Min().State(), samplingInterval)
+			defer opMin.Close()
+			if opMinN := opMin.Next(t); opMinN != nil {
+				if _, ok := opMinN.Val(); ok {
+					t.Fatalf("streaming /components/component/optical-channel/state/output-power/min is not expected to be reported")
+				}
+			}
+
+			opMax := samplestream.New(t, dut, gnmi.OC().Component(tr).OpticalChannel().OutputPower().Max().State(), samplingInterval)
+			defer opMax.Close()
+			if opMaxN := opMax.Next(t); opMaxN != nil {
+				if _, ok := opMaxN.Val(); ok {
+					t.Fatalf("streaming /components/component/optical-channel/state/output-power/max is not expected to be reported")
+				}
+			}
+
+			gnmi.Update(t, dut, gnmi.OC().Interface(dp.Name()).Enabled().Config(), bool(true))
+
+			powerStreamMap := map[string]*samplestream.SampleStream[float64]{
+				"inst": opInst,
+				"avg":  opAvg,
+				"min":  opMin,
+				"max":  opMax,
+			}
+
+			validateOutputPower(t, dut, powerStreamMap)
+
+			// Derive transceiver names from ports.
+			component := gnmi.OC().Component(tr)
+
+			outputPower := gnmi.Get(t, dut, component.OpticalChannel().TargetOutputPower().State())
+			if outputPower != targetOutputPowerdBm {
```

Comment:
you probably want to allow +/- 1dBm of deviation from the configured target power of -10dBm

Reply:
+1

Added the tolerance check for the target power. 

To be consistent I also added a tolerance for the target frequency `targetFrequencyToleranceHz    = 100000` . Please let me know if that tolerance value looks ok .